ext/featurefmt/rpm/contentmanifest/updater.go

```diff
@@ -0,0 +1,170 @@
+package contentmanifest
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/quay/clair/v3/pkg/httputil"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	lastUpdateDate  = time.Date(1970, time.January, 1, 1, 1, 1, 1, time.UTC)
+	lastHeaderQuery = time.Date(1970, time.January, 1, 1, 1, 1, 1, time.UTC)
+
+	mappingFileMutex = sync.Mutex{}
+	updaterMutex     = sync.Mutex{}
+)
+
+// LocalUpdaterJob periodically updates mapping file and store it in local storage
+type LocalUpdaterJob struct {
+	LocalPath string
+	URL       string
+}
+
+// Get translate repositories into CPEs using a mapping file
+func (updater *LocalUpdaterJob) Get(repositories []string) ([]string, error) {
+	mappingFileMutex.Lock()
```

Comment:
Why is a method on a struct utilizing a global lock? 
If the application constructs more then one "updater" then the usage of one will lock the usage of another. 
Is this intended??

Reply:
Fixed. Mutex moved into struct